lib/index.js

```diff
@@ -155,9 +133,7 @@ exports.getMockRoot = function() {
  * Restore the fs bindings for the real file system.
  */
 exports.restore = function() {
-  restoreBinding();
-  restoreProcess();
-  restoreCreateWriteStream();
+  delete realBinding._mockedBinding;
```

Comment:
We could also restore `process.cwd` and `process.chdir` to their original values here.  Then the code that overrides them would need to be moved.

Curious what you think.

Reply:
The reason I patch but removed restore is to cater `bypass()` which only touches `_mockedBinding`.

So that, all functions will behaviour like original if `_mockedBinding` is absent.

Reply:
Probably should add few unit tests to bypass to cover the edge cases on cwd/chdir/writeStream.

Reply:
Your changes here all look good to me.  And I agree it is an improvement to consistently rely on the `_mockedBinding` property.

My comment was about making `restore` actually restore `process.cwd` and `process.chdir` to their original values.  I know that they behave the same as the original functions when `realBinding._mockedBinding` is absent, but it also might be nice to have `restore` clean up all the changes it makes.  We could do this later if we find that it is actually important in some case (who knows what other libs might be expecting about these properties).  So I think this is good to merge as soon as you're comfortable with it (and agree that `bypass` could use more tests).

Reply:
Right, I missed your point. Retoring cwd/chdir doesn't break bypass. I will refactor that and add some tests.

Reply:
Cleaned up. Added one test to cover cwd/chdir.